pkg/edgeview/src/copyfile.go

```diff
@@ -316,82 +451,37 @@
 		_ = os.Remove(fileCopyDir + downloadedFile)
 	}
 
-	fileStr := strings.SplitN(downloadedFile, ".tar", 2)
-	if len(fileStr) != 2 {
-		return
-	}
-	logSaveDir := fileCopyDir + fileStr[0]
-	fmt.Printf("\n log files saved at %s\n\n", logSaveDir)
-	files, err := os.ReadDir(logSaveDir)
-	if err != nil {
-		fmt.Printf("read %s error, %v\n", logSaveDir, err)
-		return
-	}
-	for _, f := range files {
-		if info, err := f.Info(); err == nil {
-			fmt.Printf("file: %s, size %d\n", f.Name(), info.Size())
-		}
-	}
-
-	unpackLogfiles(fileCopyDir+fileStr[0], files)
-}
-
-func runCopyLogfiles(logfiles []logfiletime, time1 int64) {
-
-	timeStr := getFileTimeStr(time.Unix(time1, 0))
-	destinationfile := "/tmp/logfiles-" + timeStr + ".tar"
-
-	// no need for compression since the logfiles are already in
-	// gzip compressed format
-	tarfile, err := os.Create(destinationfile)
-	if err != nil {
-		log.Errorf("runCopyLogfiles create error %v", err)
-		return
-	}
-	defer tarfile.Close()
-
-	var fileWriter io.WriteCloser = tarfile
-
-	tarfileWriter := tar.NewWriter(fileWriter)
-	defer tarfileWriter.Close()
-
-	for _, logfile := range logfiles {
-		fileInfo, err := os.Stat(logfile.filepath)
-		if err != nil {
-			log.Errorf("runCopyLogfiles can not stat: %v", err)
-			continue
-		}
-		file, err := os.Open(logfile.filepath)
+	newlogDir := fileCopyDir + "newlog"
+	_, err = os.Stat(newlogDir)
+	if err == nil {
+		err = os.Rename(newlogDir, logSaveDir)
```

Comment:
@naiming-zededa can you come up with an approach for verifying the constructed paths to make sure the caller or user can not use ../../etc/passwd type paths. Normally path.Clean removes internal .. etc.

Once that pattern/approach is established you can apply it to this PR and review the other places in edgeview where the user/caller can feed in paths. I think they are all flagged by CodeQL.

Reply:
@eriknordmark updated the patch. made sure the directory starts with pre-defined '/download', and filepath.Clean() it to be sure it will not container '..' in the path string. Also this is inside client side of the container the volume being mount from host like /tmp/download at runtime.